lapack/gonum/lapack.go

```diff
@@ -54,4 +54,31 @@ const (
 	// drtmin is sqrt(dlamchS/dlamchP)
 	drtmin = 0x1p-485
 	drtmax = 1 / drtmin
+
+	// Blue's scaling constants
+	//
+	// An n-vector x is well-scaled if
+	//  dtsml ≤ |xᵢ| ≤ dtbig for 0 ≤ i < n and n ≤ 1/dlamchP,
+	// where
+	//  dtsml = 2^ceil((expmin-1)/2) = 2^ceil((-1021-1)/2) = 2^{-511} = 1.4916681462400413e-154
+	//  dtbig = 2^floor((expmax-digits+1)/2) = 2^floor((1024-53+1)/2) = 2^{486} = 1.997919072202235e+146
+	// If any xᵢ is not well-scaled, then multiplying small values by dssml and
+	// large values by dsbig avoids underflow or overflow when computing the sum
+	// of squares \sum_0^{n-1} (xᵢ)².
+	//  dssml = 2^{-floor((expmin-digits)/2)} = 2^{-floor((-1021-53)/2)} = 2^537 = 4.4989137945431964e+161
+	//  dsbig = 2^{-ceil((expmax+digits-1)/2)} = 2^{-ceil((1024+53-1)/2)} = 2^{-538} = 1.1113793747425387e-162
+	//
+	// References:
+	//  - Anderson E. (2017)
+	//    Algorithm 978: Safe Scaling in the Level 1 BLAS
+	//    ACM Trans Math Softw 44:1--28
+	//    https://doi.org/10.1145/3061665
+	//  - Blue, James L. (1978)
+	//    A Portable Fortran Program to Find the Euclidean Norm of a Vector
+	//    ACM Trans Math Softw 4:15--23
+	//    https://doi.org/10.1145/355769.355771
+	dtsml = 0x1p-511
+	dtbig = 0x1p486
+	dssml = 0x1p537
+	dsbig = 0x1p-538
```

Comment:
I'm happy with these (they match the [reference implementation](https://dl.acm.org/action/downloadSupplement?doi=10.1145%2F3061665&file=978.zip)).

Reply:
Ok, I'll remove the draft status.